From b7e88e081b858f963e27dad9327ecf7efa682855 Mon Sep 17 00:00:00 2001
From: Nickolay Savchenko <n.savchenko@axioma.lv>
Date: Thu, 16 Jul 2026 03:35:29 +0300
Subject: [PATCH 1/6] dt-bindings: add Ralink RT6855A and RT6856 controllers

Document the SoC, interrupt controller, GPIO, Ethernet and SPI blocks.

Also document the watchdog and system-controller compatibles.

Signed-off-by: Nickolay Savchenko <n.savchenko@axioma.lv>
---
 .../bindings/gpio/ralink,rt6856-gpio.yaml     | 32 ++++++++++++++
 .../ralink,rt6855-intc.yaml                   | 40 ++++++++++++++++++
 .../bindings/mips/ralink,rt6855.yaml          | 24 +++++++++++
 .../bindings/net/ralink,rt6855-eth.yaml       | 42 +++++++++++++++++++
 .../soc/ralink/ralink,rt6855-sysc.yaml        | 32 ++++++++++++++
 .../bindings/spi/ralink,rt6856-spi.yaml       | 30 +++++++++++++
 .../bindings/watchdog/ralink,rt6855-wdt.yaml  | 28 +++++++++++++
 7 files changed, 228 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/gpio/ralink,rt6856-gpio.yaml
 create mode 100644 Documentation/devicetree/bindings/interrupt-controller/ralink,rt6855-intc.yaml
 create mode 100644 Documentation/devicetree/bindings/mips/ralink,rt6855.yaml
 create mode 100644 Documentation/devicetree/bindings/net/ralink,rt6855-eth.yaml
 create mode 100644 Documentation/devicetree/bindings/soc/ralink/ralink,rt6855-sysc.yaml
 create mode 100644 Documentation/devicetree/bindings/spi/ralink,rt6856-spi.yaml
 create mode 100644 Documentation/devicetree/bindings/watchdog/ralink,rt6855-wdt.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/gpio/ralink,rt6856-gpio.yaml
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/gpio/ralink,rt6856-gpio.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6856 GPIO controller
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+properties:
+  compatible:
+    const: ralink,rt6856-gpio
+
+  reg:
+    maxItems: 1
+
+  gpio-controller: true
+
+  '#gpio-cells':
+    const: 2
+
+required:
+  - compatible
+  - reg
+  - gpio-controller
+  - '#gpio-cells'
+
+additionalProperties: false
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/interrupt-controller/ralink,rt6855-intc.yaml
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/interrupt-controller/ralink,rt6855-intc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6855 primary interrupt controller
+
+description: |
+  The primary interrupt-controller bank provides the 32 interrupt sources
+  used by the RT6855A and RT6856 core peripherals. The separate secondary
+  bank is not part of this binding.
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+allOf:
+  - $ref: /schemas/interrupt-controller.yaml#
+
+properties:
+  compatible:
+    const: ralink,rt6855-intc
+
+  reg:
+    maxItems: 1
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    const: 1
+
+required:
+  - compatible
+  - reg
+  - interrupt-controller
+  - '#interrupt-cells'
+
+additionalProperties: false
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/mips/ralink,rt6855.yaml
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mips/ralink,rt6855.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6855 and RT6856 based platforms
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+properties:
+  $nodename:
+    const: /
+
+  compatible:
+    items:
+      - enum:
+          - zyxel,keenetic-giga-ii
+      - const: ralink,rt6855-soc
+
+additionalProperties: true
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/ralink,rt6855-eth.yaml
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/ralink,rt6855-eth.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6855 frame engine Ethernet controller
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+allOf:
+  - $ref: ethernet-controller.yaml#
+
+properties:
+  compatible:
+    const: ralink,rt6855-eth
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  mdio-bus:
+    $ref: mdio.yaml#
+    unevaluatedProperties: false
+
+  '#address-cells':
+    const: 1
+
+  '#size-cells':
+    const: 0
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+unevaluatedProperties: false
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/ralink/ralink,rt6855-sysc.yaml
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/ralink/ralink,rt6855-sysc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6855 system, memory and PCIe reset controllers
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+properties:
+  compatible:
+    oneOf:
+      - const: ralink,rt6855-sysc
+      - items:
+          - const: ralink,rt6855-pcie-reset
+          - const: syscon
+      - items:
+          - const: ralink,rt6855-memc
+          - const: ralink,rt3050-memc
+
+  reg:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/spi/ralink,rt6856-spi.yaml
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/spi/ralink,rt6856-spi.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6856 SPI controller
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+allOf:
+  - $ref: /schemas/spi/spi-controller.yaml#
+
+properties:
+  compatible:
+    const: ralink,rt6856-spi
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - '#address-cells'
+  - '#size-cells'
+
+unevaluatedProperties: false
+
+...
--- /dev/null
+++ b/Documentation/devicetree/bindings/watchdog/ralink,rt6855-wdt.yaml
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/watchdog/ralink,rt6855-wdt.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink RT6855 watchdog timer
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+allOf:
+  - $ref: watchdog.yaml#
+
+properties:
+  compatible:
+    const: ralink,rt6855-wdt
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+  - clocks
+
+unevaluatedProperties: false
+
+...
